src/polydatum/dal.py

```diff
@@ -65,19 +107,15 @@ def __getitem__(self, path):
 
             dal['myservice.get'](my_id)
         """
+        path_segments = []
         paths = path.split('.')
-        p = paths.pop(0)
-        if p == 'dal':
-            p = paths.pop(0)
-        loc = self._services[p]
-        while 1:
-            try:
-                p = paths.pop(0)
-            except IndexError:
-                break
-            else:
-                loc = getattr(loc, p)
-        return loc
+        for p in paths:
+            if p != 'dal':
```

Comment:
Bug: Only the first `dal` path is ignored. It is possible (though not recommended) to do `dal.foo.bar.dal()`.

- [ ] Only ignore first `dal` in path.

```suggestion
paths = path.split(".")
if paths[0] == "dal":
    paths.pop(0)
path_segments = tuple(PathSegment(name=p) for p in paths)
```